### PR TITLE
Fix gaps between implementation and spec

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2479

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Review by daneshk — CHANGES_REQUESTED** [review]

File: ballerina/tests/xml_from_json_test.bal [hunk 1]
```diff
@@ -548,14 +542,95 @@ isolated function testWithAttribute1() {
...
+    }
+}
+
+@test:Config {
+    groups: ["fromJson"]
+}
+isolated function testJsonKey() {
+    json data = {"#content":"text"};
+    xml expected = xml `<root>text</root>`;
+    xml|Error? result = fromJson(data);
```
[the remark this answers is not in the document]
Comment (kalaiyarasiganeshalingam) [reply ballerina/tests/xml_from_json_test.bal:630]: When converting XML to JSON, if XML has a text node, that node will be converted with key as `#content` and value is that node value.

https://github.com/ballerina-platform/module-ballerina-xmldata/pull/348/files#diff-3922e0c8d7fa41b84add44fef7f87bb903cc9a87402f3d37317822469c705189R519

This is a reverse part of the above, so we have to create a text node but that hasn't any XML tag. So we are creating a default tag as `root`

Another test for this scenario to get more info:  https://github.com/ballerina-platform/module-ballerina-xmldata/pull/348/files#diff-6a172bf816cae1168f85b19f880a91efcd689250abc48f85e6fe68516afe7e6cR643

[the remark this answers is not in the document]
Comment (kalaiyarasiganeshalingam) [reply ballerina/tests/xml_from_json_test.bal:648]: Removed `root` tag from the output as per the offline discussion.

File: native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java [hunk 2]
```diff
@@ -284,8 +285,22 @@ private static Object convertHeterogeneousSequence(String attributePrefix, boole
         BMap<BString, Object> mapJson = newJsonMap();
         for (BXml bxml : sequence) {
             if (isCommentOrPi(bxml)) {
+                //
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java:288]: why there is no logic here

Comment (kalaiyarasiganeshalingam) [reply native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java:288]: If XML has comments or PI, we have to skip those when converting to JSON or record. So we are using this to skip those.

Comment (kalaiyarasiganeshalingam) [reply native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java]: Updated this by calling `continue`

**Review by daneshk — APPROVED** [review]
LGTM

**Pull request merged**